File: src/main/java/frc/team3683/robot/configs/java/Comp.java
package frc.team3683.robot.configs.java;

import com.ctre.phoenix.motorcontrol.can.VictorSPX;

import edu.wpi.first.wpilibj.*;
import frc.team3683.robot.utils.BNO055;

/**
 * Config
 */

/**
 * Comp
 */
public class Comp implements Config {


    public Comp() {
    
    }

	@Override
	public VictorSPX getDriveVictorLeft1() {
		return new VictorSPX(0);
	}

	@Override
	public VictorSPX getDriveVictorLeft2() {
		return new VictorSPX(1);
	}

	@Override
	public VictorSPX getDriveVictorRight1() {
		return new VictorSPX(2);
	}

	@Override
	public VictorSPX getDriveVictorRight2() {
		return new VictorSPX(3);
	}

    @Override
	public DoubleSolenoid getDriveSolenoid() {
		return new DoubleSolenoid(2, 5);
    }
    
	@Override
	public Encoder getDriveEncoderLeft() {
		return new Encoder(0, 1);
	}

	@Override
	public Encoder getDriveEncoderRight() {
		return new Encoder(2, 3);
	}

	@Override
	public BNO055 getDriveGyro() {
		return BNO055.getInstance(BNO055.opmode_t.OPERATION_MODE_IMUPLUS,
        BNO055.vector_type_t.VECTOR_EULER,
        I2C.Port.kOnboard,
        BNO055.BNO055_ADDRESS_A);
	}

	@Override
	public double getDriveP() {
		return 0;
	}

	@Override
	public double getDriveI() {
		return 0;
	}

	@Override
	public double getDriveD() {
		return 0;
	}

	@Override
	public double getGyroP() {
		return 0;
	}

	@Override
	public double getGyroI() {
		return 0;
	}

	@Override
	public double getGyroD() {
		return 0;
	}
}

File: src/main/java/frc/team3683/robot/configs/Comp.kt
package frc.team3683.robot.configs

import com.ctre.phoenix.motorcontrol.can.VictorSPX
import edu.wpi.first.wpilibj.*
import frc.team3683.robot.utils.BNO055
import frc.team3683.robot.utils.TCS34725

object Comp : Config {
    override val holoflexDown: DigitalInput = DigitalInput(7)
    override val holoflexUp: DigitalInput = DigitalInput(6)
    //drivebase
    override val driveEncoderLeft: Encoder = Encoder(0, 1)
    override val driveEncoderRight: Encoder = Encoder(2, 3)


    override val driveVictorLeft1: VictorSPX = VictorSPX(0)
    override val driveVictorLeft2: VictorSPX = VictorSPX(1)
    override val driveVictorRight1: VictorSPX = VictorSPX(2)
    override val driveVictorRight2: VictorSPX = VictorSPX(3)
    override val driveSolenoid: DoubleSolenoid = DoubleSolenoid(2, 5)
    //make 7 or 0 do someothing like on or we leak like a leak.

    override val driveTicksPerInchLeft: Double = 394.9 * 1.05
    override val driveTicksPerInchRight: Double = -395.6 * 1.05
    override val driveGyro: BNO055 = BNO055.getInstance(BNO055.opmode_t.OPERATION_MODE_IMUPLUS,
            BNO055.vector_type_t.VECTOR_EULER,
            I2C.Port.kOnboard,
            BNO055.BNO055_ADDRESS_A)

    override val colorSensor: TCS34725 = TCS34725(
            TCS34725.tcs34725IntegrationTime_t.TCS34725_INTEGRATIONTIME_24MS,
            TCS34725.tcs34725Gain_t.TCS34725_GAIN_1X,
            I2C.Port.kOnboard,
            TCS34725.TCS34725_ADDRESS_A)

    override val driveP: Double = 0.10
    override val driveI: Double = 0.000
    override val driveD: Double = 0.024
    override val gyroP: Double = 0.048
    override val gyroI: Double = 0.000012
    override val gyroD: Double = 0.0050

    //bezier
    override val xMultiplier: Double = 6.0
    override val yMultiplier: Double = 4.0

    //intake
    override val intakeVictorLeft: Victor = Victor(2)
    override val intakeVictorRight: Victor = Victor(7)
    override val wristVictor1: Victor = Victor(0)
    override val wristVictor2: Victor = Victor(1)
    //override val intakeSolenoidRight: DoubleSolenoid = DoubleSolenoid(9, 10)
    override val intakeP: Double = 0.4//4
    override val intakeI: Double = 0.0
    override val intakeD: Double = 0.08//.08
    override val Pot: AnalogPotentiometer = AnalogPotentiometer(0)
    override val LightSensor: AnalogInput = AnalogInput(1)

    override val ticksPerDegree: Double = 1/620.68
    override val zeroDegreeTicks: Double = 0.44
    //override val intakeSolenoidLeft: DoubleSolenoid = DoubleSolenoid(0, 7)
    override val LEDLeft: Solenoid = Solenoid(3)
    override val LEDRight: Solenoid = Solenoid(6)



    //elevator
    override val elevatorVictor1: Victor = Victor(3)
    override val elevatorVictor2: Victor = Victor(4)
    override val elevatorVictor3: Victor = Victor(5)
    override val elevatorVictor4: Victor = Victor(6)
    override val elevatorVictor5: Victor = Victor(8)

    override val elevatorEncoder: Encoder = Encoder(8, 9)
    override val elevatorP: Double = 0.1

    override val elevatorI: Double = 0.0000
    override val elevatorD: Double = 0.00
    override val ticksPerInch: Double = -55.0
    override val elevatorTolerance: Double = 0.5
    //override val ratchetSolenoid: DoubleSolenoid = DoubleSolenoid(1, 4)

}